src/Microsoft.Identity.Client/User.cs

```diff
@@ -47,37 +46,34 @@ internal User(User other)
             IdentityProvider = other.IdentityProvider;
         }
 
+        public User(string identifier, string displayableId, string name, string identityProvider)
+        {
+            if (string.IsNullOrWhiteSpace(identifier))
+            {
+                throw new ArgumentNullException(nameof(identifier));
+            }
+
+            DisplayableId = displayableId;
+            Name = name;
+            IdentityProvider = identityProvider;
```

Comment:
It's not just about the behaviour of acquiring a token; it's also about the data that is returned.
Put yourself if the position of a developer consuming MSAL: you call the app to acquire a token, and if the call succeeds you display the _Result.User.Name_ in the UI. How do you test that the name is being displayed correctly if you can't create a user object with valid data?

Or how do they test that their app acquires a token and then later passes the returned _Result.User_ to _IClientApplicationBase.Remove(user)_? Using a null _User_ isn't sufficient because they want to check that their application is correctly passing around the expected user object. They are not testing MSAL - they are testing that their app correctly passes around the user object that is returned by their mocked version of MSAL.